src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -40,59 +32,51 @@ void CAveragedPerceptron::init()
 	SG_ADD(&learn_rate, "learn_rate", "Learning rate.", MS_AVAILABLE);
 }
 
-bool CAveragedPerceptron::train_machine(CFeatures* data)
+void CAveragedPerceptron::train_machine(CFeatures* features, CLabels* labels)
```

Comment:
dont we want this to be const parameters?

Reply:
of course. unfortunately we can't use const for the time being because there are many non-const methods that should be const logically, e.g. `get_feature_matrix`

Reply:
I see, sure
(though get_feature_matrix can be made const without problem)